packages/pangea-sdk/pangea/services/vault/models/common.py

```diff
@@ -129,59 +153,59 @@ class CommonGenerateRequest(APIRequestModel):
     folder: Optional[str] = None
     metadata: Optional[Metadata] = None
     tags: Optional[Tags] = None
-    auto_rotate: Optional[bool] = None
-    rotation_policy: Optional[str] = None
-    store: Optional[bool] = None
+    rotation_frequency: Optional[str] = None
+    rotation_state: Optional[ItemVersionState] = None
     expiration: Optional[datetime.datetime] = None
-    managed: Optional[bool] = None
 
 
 class CommonGenerateResult(PangeaResponseResult):
     type: str
-    version: Optional[int] = None
-    id: Optional[str] = None
+    version: int
+    id: str
 
 
 class GetRequest(APIRequestModel):
     id: str
-    version: Optional[int] = None
+    version: Optional[Union[str, int]] = None
     verbose: Optional[bool] = None
+    version_state: Optional[ItemVersionState] = None
 
 
-class CommonGetResult(PangeaResponseResult):
-    type: str
-    id: str
+class ItemVersionData(PangeaResponseResult):
     version: int
-    name: Optional[str] = None
-    folder: Optional[str] = None
-    metadata: Optional[Metadata] = None
-    tags: Optional[Tags] = None
-    auto_rotate: Optional[bool] = None
-    rotation_policy: Optional[str] = None
-    last_rotated: Optional[str] = None  # TODO: should be time
-    next_rotation: Optional[str] = None  # TODO: should be time
-    expiration: Optional[str] = None  # TODO: should be time
-    created_at: Optional[str] = None  # TODO: should be time
-    revoked_at: Optional[str] = None  # TODO: should be time
-    retain_previous_version: Optional[bool] = None
+    state: str
+    created_at: str
+    destroy_at: Optional[str] = None
+    public_key: Optional[EncodedPublicKey] = None
+    secret: Optional[str] = None
 
 
-class ListItemData(APIRequestModel):
+class ItemData(PangeaResponseResult):
     id: str
     type: str
-    managed: bool
-    last_rotated: Optional[str] = None  # TODO: should be time
-    next_rotation: Optional[str] = None  # TODO: should be time
-    expiration: Optional[str] = None  # TODO: should be time
-    rotation_policy: Optional[str] = None
-    identity: str
-    version: int
+    item_state = str
+    current_version: ItemVersionData
     name: Optional[str] = None
     folder: Optional[str] = None
     metadata: Optional[Metadata] = None
     tags: Optional[Tags] = None
-    created_at: str  # TODO: should be time
-    revoked_at: Optional[str] = None  # TODO: should be time
+    rotation_frequency: Optional[str] = None
+    rotation_state: Optional[str] = None
+    last_rotated: Optional[str] = None
+    next_rotation: Optional[str] = None
+    expiration: Optional[str] = None
+    created_at: Optional[str] = None  # FIXME: Check if delete this
+    algorithm: Optional[str] = None
+    purpose: Optional[str] = None
+
+
+class GetResult(ItemData):
+    versions: List[ItemVersionData] = []
+    rotation_grace_period: Optional[str] = None
+
+
+class ListItemData(ItemData):
+    compromised_versions: List[ItemVersionData]
```

Comment:
Where are all the missing fields?

	ID                string            `json:"id,omitempty"`
	CreatedAt         time.Time         `json:"created_at,omitempty"`
	Tags              []string          `json:"tags,omitempty"`
	Metadata          pangeadb.Metadata `json:"metadata,omitempty"`
	LastRotated       *time.Time        `json:"last_rotated,omitempty"`
	NextRotation      *time.Time        `json:"next_rotation,omitempty"`
	Expiration        *time.Time        `json:"expiration,omitempty"`
	RotationFrequency *string           `json:"rotation_frequency,omitempty"`
	RotationState     *string           `json:"rotation_state,omitempty"`
	ItemState         string            `json:"item_state"`

	// only valid for keys
	Algorithm *string `json:"algorithm,omitempty"`
	Purpose   *string `json:"purpose,omitempty"`

	CurrentVersion      *ListResponseItemVersion   `json:"current_version"`
	CompromisedVersions []*ListResponseItemVersion `json:"compromised_versions"`

Reply:
They are on ItemData

Reply:
Ok